src/smolagents/local_python_executor.py

```diff
@@ -1229,8 +1229,20 @@ def evaluate_ast(
         # For loop -> execute the loop
         return evaluate_for(expression, *common_params)
     elif isinstance(expression, ast.FormattedValue):
-        # Formatted value (part of f-string) -> evaluate the content and return
-        return evaluate_ast(expression.value, *common_params)
+        # Formatted value (part of f-string) -> evaluate the content and format it
+        value = evaluate_ast(expression.value, *common_params)
+        # Get the format spec if it exists
+        format_spec = None
+        if expression.format_spec:
+            format_spec = (
+                "".join(str(evaluate_ast(v, *common_params)) for v in expression.format_spec.values)
+                if isinstance(expression.format_spec, ast.JoinedStr)
+                else expression.format_spec.value
```

Comment:
Sorry, I don't understand when `expression.format_spec` can be not instance of `ast.JoinedStr`. Could you give an example (and maybe a test)?

Reply:
that was some sort of safe-guarding from my side, and as I looked now closely into different cases it never goes the "else" flow.

changed this part to avoid thinking about the actual expression